src/ui/router/ApplicationRouterSpec.js

```diff
@@ -1,139 +1,86 @@
     });
 
-    afterAll(() => {
-        openmct.router.setHash(initialHash);
+    afterEach(() => {
         appHolder.remove();
 
         return resetApplicationState(openmct);
     });
 
-    it('has initial hash when loaded', (done) => {
-        let success;
-        resolveFunction = () => {
-            openmct.router.setLocationFromUrl();
-            success = window.location.hash !== null;
-            if (success) {
-                initialHash = window.location.hash;
-                expect(success).toBe(true);
-
-                openmct.router.removeListener('change:hash', resolveFunction);
-                done();
-            }
-        };
-
-        openmct.router.on('change:hash', resolveFunction);
+    it('has initial hash when loaded', () => {
+        expect(window.location.hash).not.toBe(null);
     });
 
-    it('The setSearchParam function sets an individual search parameter in the window location hash', (done) => {
-        let success;
+    it('The setSearchParam function sets an individual search parameter in the window location hash', () => {
         openmct.router.setSearchParam('testParam', 'testValue');
-        resolveFunction = () => {
-            success = window.location.hash.includes('testParam=testValue');
-            if (success) {
-                expect(success).toBe(true);
-
-                openmct.router.removeListener('change:hash', resolveFunction);
-                done();
-            }
-        };
 
-        openmct.router.on('change:hash', resolveFunction);
+        const searchParams = openmct.router.getAllSearchParams();
+        expect(searchParams.get('testParam')).toBe('testValue');
     });
 
-    it('The getSearchParam function returns the value of an individual search paramater in the window location hash', () => {
-        expect(openmct.router.getSearchParam('testParam')).toBe('testValue');
-    });
+    it('The deleteSearchParam function deletes an individual search paramater in the window location hash', () => {
+        openmct.router.setSearchParam('testParam', 'testValue');
 
-    it('The deleteSearchParam function deletes an individual search paramater in the window location hash', (done) => {
-        let success;
-        openmct.router.deleteSearchParam('testParam');
-        resolveFunction = () => {
-            success = window.location.hash.includes('testParam=testValue') === false;
-            if (success) {
-                expect(success).toBe(true);
+        let searchParams = openmct.router.getAllSearchParams();
+        expect(searchParams.get('testParam')).toBe('testValue');
 
-                openmct.router.removeListener('change:hash', resolveFunction);
-                done();
-            }
-        };
+        openmct.router.deleteSearchParam('testParam');
 
-        openmct.router.on('change:hash', resolveFunction);
+        searchParams = openmct.router.getAllSearchParams();
+        expect(searchParams.get('testParam')).toBe(null);
```

Comment:
This is not testing what it says it's testing. It should be setting router params and then checking window location, not the router parameters.